src/zenml/artifact_stores/base_artifact_store.py

```diff
@@ -380,30 +458,36 @@ def _register(self) -> None:
         if isinstance(self, LocalFilesystem):
             return
 
+        overloads: Dict[str, Any] = {
+            "SUPPORTED_SCHEMES": self.config.SUPPORTED_SCHEMES,
+        }
+        for abc_method in inspect.getmembers(BaseArtifactStore):
+            if getattr(abc_method[1], "__isabstractmethod__", False):
+                overloads[abc_method[0]] = staticmethod(
+                    _sanitize_paths(self.path)(
+                        getattr(self, abc_method[0]),
+                    )
+                )
```

Comment:
I was wrong, this won't work in the local artifact store case, because this code is skipped. I think you just have to split this into two parts then. I'll push another comment with the updated code.

Reply:
Yeah, exactly - LocalFS is the trickiest part of this, since it is also used transparently in many places (working with configs and so on)